libcobblersignatures/models/osversion.py

```diff
@@ -67,124 +67,134 @@ def __eq__(self, other):
             and self.boot_loaders == other.boot_loaders
 
     @property
-    def signatures(self):
+    def signatures(self) -> list:
         return self._signatures
 
     @signatures.setter
-    def signatures(self, value):
-        self._signatures = value
+    def signatures(self, value: list):
+        if isinstance(value, list):
+            self._signatures = value
 
     @signatures.deleter
     def signatures(self):
-        del self._signatures
+        self._signatures = []
 
     @property
-    def version_file(self):
+    def version_file(self) -> str:
         return self._version_file
 
     @version_file.setter
     def version_file(self, value):
-        self._version_file = value
+        if isinstance(value, str):
+            self._version_file = value
 
     @version_file.deleter
     def version_file(self):
-        del self._version_file
+        self._version_file = ""
 
     @property
-    def version_file_regex(self):
+    def version_file_regex(self) -> str:
         return self._version_file_regex
 
     @version_file_regex.setter
     def version_file_regex(self, value):
-        self._version_file_regex = value
+        if isinstance(value, str):
+            self._version_file_regex = value
 
     @version_file_regex.deleter
     def version_file_regex(self):
-        del self._version_file_regex
+        self._version_file_regex = ""
 
     @property
-    def kernel_arch(self):
+    def kernel_arch(self) -> str:
         return self._kernel_arch
 
     @kernel_arch.setter
     def kernel_arch(self, value):
-        self._kernel_arch = value
+        if isinstance(value, str):
+            self._kernel_arch = value
```

Comment:
This is silently not setting the value if it's not a string.

Reply:
Now we raise a TypeError()